packages/runtime/container-runtime/src/summaryStorageCompressionHooks.ts

```diff
@@ -0,0 +1,199 @@
+    from "@fluidframework/protocol-definitions";
+import { IsoBuffer, Uint8ArrayToString } from "@fluidframework/common-utils";
+// import { getBlobAtPath, listBlobPaths, replaceSummaryObject, SummaryStorageHooks } from "./summaryStorageAdapter";
+import { SummaryStorageHooks } from "./summaryStorageAdapter";
+import { BlobHeaderBuilder, readBlobHeader, skipHeader, writeBlobHeader } from "./summaryBlobProtocol";
+import { SummaryCompressionAlgorithms } from "./containerRuntime";
+
+const algorithmKey = "ALG";
+const defaultMinSizeToCompress = 500;
+
+/**
+ * This class implements the SummaryStorageHooks which can apply various kinds of compressions
+ * to the blob payload.
+ */
+export class CompressionSummaryStorageHooks implements SummaryStorageHooks {
+    public get algorithm() {
+        return this._algorithm;
+    }
+    private readonly blobReplacer = (input: any, context: any) => {
+        if (input.type === SummaryType.Blob) {
+            const summaryBlob: ISummaryBlob = input;
+            const decompressed: Uint8Array = typeof summaryBlob.content === "string" ?
+                new TextEncoder().encode(summaryBlob.content) : summaryBlob.content;
+            if (this._minSizeToCompress !== undefined && decompressed.length < this._minSizeToCompress) {
+                return input;
+            }
+            const compressed: ArrayBufferLike = this.encodeBlob(decompressed);
+            let newSummaryBlob;
+            if (this._isUseB64OnCompressed !== undefined && this._isUseB64OnCompressed) {
+                // TODO: This step is now needed, it looks like the function summaryTreeUploadManager#writeSummaryBlob
+                // fails on assertion at 2 different generations of the hash which do not lead to
+                // the same result if the ISummaryBlob.content is in the form of ArrayBufferLike
+                const compressedString = Uint8ArrayToString(IsoBuffer.from(compressed), "base64");
+                const compressedEncoded = new TextEncoder().encode(compressedString);
+                newSummaryBlob = { type: SummaryType.Blob, content: compressedEncoded };
+            } else {
+            // This line will replace the 3 lines above when the bug is fixed.
+            // const newSummaryBlob: ISummaryBlob = { type: SummaryType.Blob, content: IsoBuffer.from(compressed)};
+                newSummaryBlob = { type: SummaryType.Blob, content: IsoBuffer.from(compressed) };
+            }
+            return newSummaryBlob;
+        } else {
+            return input;
+        }
+    };
+    constructor(private readonly _algorithm: SummaryCompressionAlgorithms | undefined,
+        private readonly _minSizeToCompress: number | undefined,
+        private readonly _isUseB64OnCompressed: boolean | undefined) {
+            if (this._minSizeToCompress === undefined) {
+                this._minSizeToCompress = defaultMinSizeToCompress;
+            }
+        }
```

Comment:
Nit: mismatched indentation.

It helps if you move the closing paren/brace to its own line like this:
```ts
constructor(
    ...
) {
    ...
}
```